drake/core/test/matrix_compare_test.cc

```diff
@@ -0,0 +1,152 @@
+#include "drake/util/eigen_matrix_compare.h"
+#include "gtest/gtest.h"
+
+using drake::util::MatrixCompareType;
+
+namespace drake {
+namespace test {
```

Comment:
I didn't want to pollute the `drake::util` namespace and thought "test" better describes the purpose of this class compared to an anonymous namespace. Should I use three levels of namespaces instead?
```
namespace drake {
namespace util {
namespace test {
```